source/simulator/assembly.cc

```diff
@@ -47,17 +48,24 @@ namespace aspect
       namespace Scratch
       {
         template <int dim>
+
+
+
+
         StokesPreconditioner<dim>::
         StokesPreconditioner (const FiniteElement<dim> &finite_element,
                               const Quadrature<dim>    &quadrature,
                               const Mapping<dim>       &mapping,
                               const UpdateFlags         update_flags,
-                              const unsigned int        n_compositional_fields)
+                              const unsigned int        n_compositional_fields,
+                              const bool                add_compaction_pressure)
           :
           finite_element_values (mapping, finite_element, quadrature,
                                  update_flags),
           grads_phi_u (finite_element.dofs_per_cell, Utilities::signaling_nan<SymmetricTensor<2,dim> >()),
           phi_p (finite_element.dofs_per_cell, Utilities::signaling_nan<double>()),
+          phi_p_c (add_compaction_pressure ? finite_element.dofs_per_cell : 0),
```

Comment:
everything else is initialized with signaling_nans, can you do that for `p_c` and `grad_phi_p` as well?